components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -56,6 +60,60 @@ CurlDownloadHandler::CurlDownloadHandler(
         m_http_headers.append("Cache-Control: no-cache");
         m_http_headers.append("Pragma: no-cache");
     }
+    if (http_header_kv_pairs.has_value()) {
+        constexpr std::array<std::string_view, 3> cReservedHeaders
```

Comment:
How about:
1. Traverse the reversed-key array and throw if any reversed key exists in the given kv pairs
2. Then traverse the kv-pairs to construct curl strings, and validate the curl string is not CRLF terminated

Reasons:
1. Currently, we compare all header keys with the all reserved keys (in the best case), but we only need three comparisons for the reserved keys.
2. Splitting reversed key check and the actual header string constructions won't introduce any overhead